src/openvslam_ros.cc

```diff
@@ -86,35 +89,68 @@ void system::setParams() {
 
 void system::init_pose_callback(
     const geometry_msgs::msg::PoseWithCovarianceStamped::SharedPtr msg) {
-    Eigen::Matrix4d cam_pose_ros(Eigen::Matrix4d::Identity());
-    Eigen::Vector3d pose_v(
+    if (camera_link_.empty()) {
+        RCLCPP_ERROR(node_->get_logger(),
+                     "Camera link is not set: no images were received yet");
+        return;
+    }
+
+    Eigen::Translation3d trans(
         msg->pose.pose.position.x,
         msg->pose.pose.position.y,
         msg->pose.pose.position.z);
-    cam_pose_ros.block<3, 1>(0, 3) = pose_v;
-
     Eigen::Quaterniond rot_q(
         msg->pose.pose.orientation.w,
         msg->pose.pose.orientation.x,
         msg->pose.pose.orientation.y,
         msg->pose.pose.orientation.z);
-    Eigen::Matrix3d rot_m = rot_q.toRotationMatrix();
-    cam_pose_ros.block<3, 3>(0, 0) = rot_m;
-
-    Eigen::Matrix4d rot_ros_to_cv_map_frame;
-    rot_ros_to_cv_map_frame << 0, 0, 1, 0,
-        -1, 0, 0, 0,
-        0, -1, 0, 0,
-        0, 0, 0, 1;
-
-    // Transforming initial pose from ROS coordinate system to CV coordinate system as follows:
-    // T(map, pose_cv) = T(map, pose) * T(pose, pose_cv)
-    // T(map, pose_cv) = T(map, map_cv) * T(map_cv, pose_cv)
-    // where T(pose, pose_cv) == T(map, map_cv)
-    // ||
-    // \/
-    // T(map_cv, pose_cv) = T(map, map_cv).inv * T(map, pose) * T(map, map_cv)
-    Eigen::Matrix4d cam_pose_cv = rot_ros_to_cv_map_frame.transpose() * cam_pose_ros * rot_ros_to_cv_map_frame;
+    Eigen::Affine3d initialpose_affine(trans * rot_q);
+
+    Eigen::Matrix3d rot_cv_to_ros_map_frame;
+    rot_cv_to_ros_map_frame << 0, -1, 0,
+        0, 0, -1,
+        1, 0, 0;
+
+    Eigen::Affine3d map_to_initialpose_frame_affine;
+    if (msg->header.frame_id == map_frame_) {
+        map_to_initialpose_frame_affine = Eigen::Affine3d::Identity();
+    }
+    else {
```

Comment:
It looks unnecessary.

Reply:
Since in most cases the `frame_id` where initial pose is requested will be equal to `map_frame_`, we do not need to do all this `tf->lookupTransform` / `transformToEigen` conversion stuff. So, that is some kind of optimization applicable on that.

Or do you mean to initialize `map_to_initialpose_frame_affine` with identical transform?
```
Eigen::Affine3d map_to_initialpose_frame_affine(Eigen::Affine3d::Identity());
if (msg->header.frame_id != map_frame_) {
    try {
        auto map_to_initialpose_frame = tf_->lookupTransform(
            map_frame_, msg->header.frame_id, tf2_ros::fromMsg(msg->header.stamp),
            tf2::durationFromSec(0.0));
            ...
```

Reply:
No optimization is required here. Give priority to simplicity.

Reply:
Done in https://github.com/OpenVSLAM-Community/openvslam_ros/pull/29/commits/ae2e047e905892a3c332f018239bf4f938096a23. However, I still doubt that there is something complex to refuse it.